cpp/lib/src/Elasticsearch.cc

```diff
@@ -58,28 +70,139 @@ Elasticsearch::Elasticsearch(const std::string &ini_file_path) {
 //    We know that there is an _update API endpoint, but as we insert a bunch of chunks, the number of which can change,
 //    we have to use a "wildcard" strategy to first delete possibly existing chunks as the number of new chunks may be greater or
 //    fewer than what may already be stored.
-void Elasticsearch::insertDocument(const std::string &document_id, const std::vector<std::string> &document_chunks) {
+void Elasticsearch::insertDocument(const std::string &document_id, const std::string &document) {
```

Comment:
insertDocument call deleteDocument, so please either change the name of the member function or change documentation